File: booking_service/src/infrastructure/repository/mapper/accommodation_mapper.go
package mapper

import (
	"booking_service/core/domain/entity"
	"booking_service/infrastructure/repository/model"
)

func ToAccommodationEntity(accommodation *model.AccommodationModel) *entity.AccommodationEntity {
	return &entity.AccommodationEntity{
		BaseEntity: entity.BaseEntity{
			ID:        accommodation.ID,
			CreatedAt: accommodation.CreatedAt.Unix(),
			UpdatedAt: accommodation.UpdatedAt.Unix(),
		},
		Name: accommodation.Name,
	}
}

func ToAccommodationModel(accommodation *entity.AccommodationEntity) *model.AccommodationModel {
	return &model.AccommodationModel{
		BaseModel: model.BaseModel{
			ID: accommodation.ID,
		},
		Name: accommodation.Name,
	}
}

File: accommodation_service/src/main/java/huy/project/accommodation_service/infrastructure/repository/mapper/AccommodationMapper.java
package huy.project.accommodation_service.infrastructure.repository.mapper;

import huy.project.accommodation_service.core.domain.entity.AccommodationEntity;
import huy.project.accommodation_service.infrastructure.repository.model.AccommodationModel;
import org.mapstruct.Mapper;
import org.mapstruct.factory.Mappers;

import java.util.List;

@Mapper
public abstract class AccommodationMapper {
    public static final AccommodationMapper INSTANCE = Mappers.getMapper(AccommodationMapper.class);

    public abstract AccommodationEntity toEntity(AccommodationModel model);

    public abstract AccommodationModel toModel(AccommodationEntity entity);

    public abstract List<AccommodationEntity> toListEntity(List<AccommodationModel> models);
}
